scripts/utils/bdlClassnameManager

```diff
@@ -0,0 +1,225 @@
+    // argv 2 is the operation: check, swap, append, extend
+    const operation = process.argv[2];
+
+    // argv 3 is the file (and extension)
+    const fileName = process.argv[3];
+
+    if (!fileName) {
+        console.error(colors.red('Missing parameter:'), colors.white('fileName'));
+        process.exit(1);
+    }
+    if (!Object.keys(conversionMap).length) {
+        console.error(colors.red('Missing values in: '), colors.white('conversionMap'));
+        process.exit(1);
+    }
+    const fileType = fileName.split('.').pop();
+    const verboseMode = process.argv[4] === '--verbose';
+    const extendMode = operation === 'extend'
+    const currentConversionMap = extendMode ? invert(conversionMap) : conversionMap;
+
+    let reBadName;
+    if (fileType === 'scss') {
+        const converter = entry => `\\${scssPrefix}${entry}(?!-)`;
+        reBadName = new RegExp(
+            Object.keys(currentConversionMap)
+                .map(converter)
+                .sort(lengthSorter) // sort order matters here for regex
+                .join('|'), 'g'
+        );
+    } else if (fileType === 'js') {
+        // The majority of .js classnames are surrounded by quotation marks
+        // Restrict to those prefixes to avoid overrwriting variable names
+        const converter = entry => jsPrefixes.map(prefix => `${prefix}${entry}`).join('|');
+        reBadName = new RegExp(Object.keys(currentConversionMap).map(converter).join('|'), 'g');
+    } else {
+        console.error(colors.yellow('Unrecognized file type for this tool. skipping', fileName, '...'));
+        process.exit(0);
+    }
+
+    try {
+        let fileToParse = await readFile(fileName, { encoding: 'utf8' });
+
+        switch (operation) {
+            case 'check':
+                // determine if the file is using one of the legacy variable names
+                const foundMatch = reBadName.test(fileToParse);
+                if (foundMatch) {
+                    // Exit with an error code: we should not find bad names
+                    if (verboseMode) {
+                        console.error(colors.red('Bad BDL class name found in', fileName));
+                    }
+                    process.exit(1);
+                }
+
+                break;
+            case 'append': {
+                const matches = [...new Set(fileToParse.match(reBadName))];
+
+                if (verboseMode) {
+                    printVerboseOperationMessage(fileName, matches);
+                }
+
+                matches
+                    .sort(lengthSorter) // sort prevents root classnames ('btn') from preemptively changing longer classnames ('btn-primary')
+                    .forEach((match) => {
+                        if (fileType === 'scss') {
+                            fileToParse = fileToParse.replace(new RegExp(`\\${match}(?!-)`, 'g'), `${scssPrefix}${currentConversionMap[kebabCase(match)]}`);
```

Comment:
Is the `kebab-case` necessary here? It seems like the source class name is likely already in that format (`btn-primary`, for example).

Reply:
True `kebab-case` isn't needed, I'll replace it with lodash's `trimStart`. The issue here is that `match` is `.tooltip` instead of `tooltip`, which is the key in the conversionMap.